autorest/codegen/__init__.py

```diff
@@ -76,6 +76,21 @@ def _validate_code_model_options(options: Dict[str, Any]) -> None:
             "If you want operation files, pass in flag --show-operations"
         )
 
+    if options["package_mode"] == 'dataplane' and \
```

Comment:
The value "dataplane" should not be hardcoded, we should have a map of value to folder, and the template folder should be part of the autorest delivery

For instance, the folder `autorest/codegen/packaging_template/dataplane` contains the `dataplane` template, and the folder `autorest/codegen/packaging_template/abcxyz` will contain the template `abcxyz`

Reply:
template could be distinguished with different folder:
![image](https://user-images.githubusercontent.com/70930885/151644540-3272852c-ada6-495e-900d-04fe0c7e6bca.png). So we could make a map with that.

And different `package-mode` may need different parameters, so here we had better keep it.